### PR TITLE
Support for debian-jessie on aarch64

### PR DESCRIPTION
**Pull request by docbobo**
This is similar to PR #30168. It does use a golang package from jessie-backports instead of a custom build one though. 

**- What I did**
- Added build of package for `debian-jessie` on `aarch64`. 

**- How I did it**
- Added `debian-jessie` to `generate.sh` script. In order to properly build Go 1.7.5 from source, `golang-1.6-go` from jessie-backports is required.

**- How to verify it**
You just need to build the Debian packages on an aarch64. As mentioned in PR #30168, this can be done with 2A machines at packet.net.

**- Description for the changelog**
Added support for debian-jessie as `make deb` output for aarch64.

### CODE REVIEW

**Comment by docbobo** [comment]
@vielmetti If I understand your correctly, all Go versions in this PR should be updated to 1.8.0, right?

**Comment by vielmetti** [comment]
@docbobo - I know that all of the mainline Docker versions are at 1.7.5 now

The alternative is to apply this backported patch to the Go sources to address the problem with page sizes:

https://github.com/fancybits/go/compare/release-branch.go1.7...arm-pagesize-64k

All that said, I am good with going to Go 1.8.0 release (and then dealing with the consequences, if any).

**Comment by docbobo** [comment]
Okay, I've updated the PR. I am also building on packet.net myself right now.

**Comment by tophj-ibm** [comment]
IMO this should be two PRs, one for the fix for manpages + validate which is more of a needed bug fix, and another for adding in debian:jessie support.

cc @justincormack although this doesn't have seccomp support so I know you aren't going to like it 😄

**Comment by docbobo** [comment]
@tophj-ibm are you also ok with raising GO_VERSION to 1.8.0?

**Comment by tophj-ibm** [comment]
@docbobo I would open a new one for the dockerfile.aarch64 and the man/dockerfile.aarch64 change. And then leave this one as all the contrib/* changes. That way #31061 can be fixed ASAP.

I would leave the GO_VERSION in dockerfile.aarch64 and in the manpages as 1.7.5 for now just so it's consistent across all the GO_VERSIONS and there is already a PR that updates them #29030. It's probably fine to leave it as 1.8 in the actual contrib files though, because that's more of a feature request.

**Comment by justincormack** [comment]
There are issues that need to be fixed before anything updates to Go 1.8 so
you need to wait for the overall PR to be merged.

**Comment by docbobo** [comment]
Okay. I've update this PR:

- No Go 1.8 for now, everything stays at 1.7.5
- Removed everything regarding golint, manpages, etc.

So this PR just contains the changes needed to support debian:jessie. As described, this is done by bootstraping Go 1.7.5 via `golang-1.6-go`. Let me know if there's anything else.

**Comment by docbobo** [comment]
Anyone having an idea how my change could've affected the Windows build? Flaky test?

**Comment by DieterReuter** [comment]
I just tested this PR on a Packet.net 2A server with building Docker v1.13.1, works like a charme!
:+1:

**Comment by vieux** [comment]
Are we good to merge this one ?

**Comment by tophj-ibm** [comment]
@justincormack thoughts on this seeing as jessie doesn't have the proper seccomp support? 

On the other hand x86 jessie doesn't have it either, so a reasonable solution might be to require it in all versions later than this, wdyt.

**Comment by justincormack** [comment]
I don't really see why anyone would sanely run `jessie` on arm64, we should at least start with `stretch` which would be a plausible install target.

`libseccomp` 2.2.3 is available in jessie backports - again if you are foolish enough to run jessie on arm64 you should have backports enabled or nothing is going to work well.

**Comment by docbobo** [comment]
@justincormack I just looked at the other builders for debian packages. Here's the current state:

* aarch64: ubuntu:trusty without seccomp.
* amd64: debian:jessie, debian:wheezy without seccomp. ubuntu:precis, ubuntu:trusty without seccomp.
* armhf: debian:jessie, raspbian:jessie without seccomp. ubuntu:trusty without seccomp.
* ppc64le: ubuntu:trusty without seccomp.
* s390x: ubuntu:xenial without seccomp.

None of the other architectures currently has a debian:jessie package with seccomp support. From that perspective, I find the PR in line with the other architectures. 

How about this: I'll do another PR for debian:strech (aarch64) and we keep debian:jessie aligned with the already existing builders?

**Comment by justincormack** [comment]
@docbobo I guess we can say we arent adding another version without seccomp as it is still just jessie, ok. Thanks for adding stretch, thats what people should be running...

The CI failures are unrelated, will do a rebuild once the fix is merged to master.

**Comment by coolljt0725** [comment]
ping @tianon

File: contrib/builder/deb/aarch64/generate.sh
Comment (tianon) [inline contrib/builder/deb/aarch64/generate.sh]: This block appears to be indented with spaces instead of tabs (which is what the rest of this file uses), although should probably just be removed instead along with the other `jessie` references per @justincormack's comment.

**Review by tianon — COMMENTED** [review]
Generally seems OK to me, but I agree with @justincormack that `jessie` probably ought to be removed for `aarch64`. :+1:

File: contrib/builder/deb/aarch64/generate.sh
Comment (docbobo) [inline contrib/builder/deb/aarch64/generate.sh]: Please help me out here. Stretch is currently in "testing" (according to https://wiki.debian.org/DebianStretch). Why exactly is this preferable over Jessie?

Comment (tianon) [inline contrib/builder/deb/aarch64/generate.sh]: Stretch is frozen, meaning no new packages without an explicit unblock from the release team (in preparation for Stretch becoming the next stable release), so it's a pretty safe target these days, but IMO more importantly, Jessie on aarch64 is not a great/safe experience (as noted in the discussion above).

Seccomp is a really important part of Docker's overall security model, so I'm definitely +1 on @justincormack's suggestion that we not add any more platforms for which it cannot be supported, and would even be +1 on either finding a way to get `libseccomp` working for them or deprecating support for them too (via the normal project-wide deprecation process of several releases).

Comment (docbobo) [inline contrib/builder/deb/aarch64/generate.sh]: ok. adding ```libseccomp``` support for jessie is rather straightforward, and I have that change done (just need to reverify). I'd rather add that than completely removing jessie.

File: contrib/builder/deb/aarch64/debian-jessie/Dockerfile
Comment (tophj-ibm) [inline contrib/builder/deb/aarch64/debian-jessie/Dockerfile]: don't forget the tags

Comment (docbobo) [inline contrib/builder/deb/aarch64/debian-jessie/Dockerfile]: argh. that was supposed to go to my testing branch first. Let me fix that quickly.

**Comment by docbobo** [comment]
```
$ docker info
Containers: 3
 Running: 0
 Paused: 0
 Stopped: 3
Images: 43
Server Version: 17.04.0-dev
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 43
 Dirperm1 Supported: true
Logging Driver: journald
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: 
containerd version: 665e84e6c28653a9c29a6db601636a92d46896f3
runc version: a01dafd48bc1c7cc12bdb01206f9fea7dd6feb70
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 3.14.79-108
Operating System: Debian GNU/Linux 8 (jessie)
OSType: linux
Architecture: aarch64
CPUs: 4
Total Memory: 1.928GiB
Name: dockeroid01
ID: SLSX:MAEN:2PA4:LETD:C36P:YAZK:W5BA:OLE7:QHHS:JOWV:BKAJ:Z2PZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Comment by docbobo** [comment]
I am having a hard time understanding why the test were failing (and why two builder seem to be stuck). Were those test failures legit? Is there any way to restart tests?

**Comment by justincormack** [comment]
rebuilding janky to see

**Comment by justincormack** [comment]
Jessie with compulsory backports seems more sane (but I still recommend people use `stretch`!)

**Review by justincormack — APPROVED** [review]

**Comment by justincormack** [comment]
ok tests passed, LGTM

**Comment by tophj-ibm** [comment]
P+Z tests should be fine too

**Review by tophj-ibm — APPROVED** [review]

**Review by tianon — APPROVED** [review]
LGTM

@thaJeztah is there someone we need to ping to make sure the "compulsory backports on jessie" gets documented appropriately?

**Comment by DieterReuter** [comment]
Cool. I really love this implementation. So we do have support for Stretch and Jessie. Great work @docbobo!

**Comment by justincormack** [comment]
ok, old P+Z builds from the jenkins update, ignoring

**Pull request merged**
